src/main/java/FolderScanner.java

```diff
@@ -0,0 +1,28 @@
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FolderScanner {
+    public static List<File> scanMp3(String folderName) {
+        List<File> mp3Files = new ArrayList<>();
+        scanMp3(new File(folderName), mp3Files);
+        return mp3Files;
+    }
+
+    private static void scanMp3(File folder, List<File> mp3Files) {
+        File[] files = folder.listFiles();
+        if (files != null) {
```

Comment:
You may invert your if statement and make it `if (files == null) return;` to avoid deep nesting